sale_manual_delivery/wizard/manual_delivery.py

```diff
@@ -0,0 +1,95 @@
+        sale_lines = self.env["sale.order.line"]
+        active_model = self.env.context["active_model"]
+        if active_model == "sale.order.line":
+            sale_ids = self.env.context["active_ids"] or []
+            sale_lines = self.env["sale.order.line"].browse(sale_ids)
+        elif active_model == "sale.order":
+            sale_ids = self.env.context["active_ids"] or []
+            sale_lines = self.env["sale.order"].browse(sale_ids).mapped("order_line")
+        if len(sale_lines.mapped("order_id.partner_id")) > 1:
+            raise UserError(_("Please select one partner at a time"))
+        if sale_lines:
+            # Get partner from those lines
+            partner = sale_lines.mapped("order_id.partner_id")
+            res["partner_id"] = partner.id
+            res["commercial_partner_id"] = partner.commercial_partner_id.id
+            # Convert to manual.delivery.lines
+            res["line_ids"] = [
+                (
+                    0,
+                    0,
+                    {
+                        "order_line_id": line.id,
+                        "name": line.name,
+                        "product_id": line.product_id.id,
+                        "qty_ordered": line.product_uom_qty,
+                        "qty_procured": line.qty_procured,
+                        "quantity": line.qty_to_procure,
+                    },
+                )
+                for line in sale_lines
+                if line.qty_to_procure and line.product_id.type != "service"
+            ]
+        return res
+
+    commercial_partner_id = fields.Many2one(
+        "res.partner",
+        required=True,
+        readonly=True,
+        ondelete="cascade",
+    )
+    partner_id = fields.Many2one(
+        "res.partner",
+        string="Delivery Address",
+        domain="""
+            [
+                "|",
+                ("id", "=", commercial_partner_id),
+                ("parent_id", "=", commercial_partner_id),
+            ],
+        """,
+        ondelete="cascade",
+    )
+    carrier_id = fields.Many2one(
+        "delivery.carrier",
+        string="Delivery Method",
+        ondelete="cascade",
+    )
+    route_id = fields.Many2one(
+        "stock.location.route",
+        string="Use specific Route",
+        domain=[("sale_selectable", "=", True)],
+        ondelete="cascade",
+        help="Leave it blank to use the same route that is in the sale line",
+    )
+    line_ids = fields.One2many(
+        "manual.delivery.line",
+        "manual_delivery_id",
+        string="Lines to validate",
+    )
+    date_planned = fields.Datetime(string="Date Planned")
+
+    def confirm(self):
+        """Creates the manual procurements"""
+        self.ensure_one()
+        sale_order_lines = self.line_ids.mapped("order_line_id")
+        sale_order_lines.with_context(
+            sale_manual_delivery=self
+        )._action_launch_stock_rule_manual()
```

Comment:
From a functional user perspective: Here can we return to the picking form like in purchase_manual_delivery? https://github.com/OCA/purchase-workflow/blob/874c7d7acbac075481a1a41578cb37beb0c7eaa0/purchase_manual_delivery/wizard/create_manual_stock_picking.py#L158

Reply:
It seems to me that this feature is more difficult to implement here, because the picking is not built in this context, the stock rules will take care of it and the result can be very varied, several pickings, it seems to me that it would be better to do an independent implementation for that case, in the other module it is more precise because the picking is created by the same assistant

Reply:
Thanks for your explanation. IMHO maybe it more ux friendly to return something, notify user that draft delivery has been created, or simply to the picking list.